lib/Sema/CSDiagnostics.cpp

```diff
@@ -1961,6 +1961,30 @@ bool MissingMemberFailure::diagnoseAsError() {
   return true;
 }
 
+bool InvalidMemberRefOnExistential::diagnoseAsError() {
+  auto *anchor = getRawAnchor();
+
+  Expr *baseExpr = getAnchor();
+  DeclNameLoc nameLoc;
+  if (auto *UDE = dyn_cast<UnresolvedDotExpr>(anchor)) {
+    baseExpr = UDE->getBase();
+    nameLoc = UDE->getNameLoc();
+  } else if (auto *UME = dyn_cast<UnresolvedMemberExpr>(anchor)) {
+    nameLoc = UME->getNameLoc();
+  } else if (auto *SE = dyn_cast<SubscriptExpr>(anchor)) {
+    baseExpr = SE->getBase();
+  } else if (auto *call = dyn_cast<CallExpr>(anchor)) {
+    baseExpr = call->getFn();
+  }
+
+  emitDiagnostic(getAnchor()->getLoc(),
+                 diag::could_not_use_member_on_existential, getBaseType()->getRValueType(),
```

Comment:
Nit: I think it makes sense to `->getRValueType()` inside of `getBaseType()` itself.